### PR TITLE
Exception fix for malformed urls to url.parse

### LINKED ISSUE #1864: url.parse exception within redirect.js
I am getting the following exception when calling request(URL,function(err,res,body){});

```
URIError: URI malformed at decodeURIComponent (native) at Url.parse (url.js:185:19) at Object.urlParse [as parse] (url.js:101:5) at Redirect.onResponse (/myproject/node_modules/request/lib/redirect.js:104:21) at Request.onRequestResponse (/myproject/node_modules/request/request.js:944:22) at ClientRequest.emit (events.js:95:17) at HTTPParser.parserOnIncomingClient [as onIncoming] (http.js:1692:21) at HTTPParser.parserOnHeadersComplete [as onHeadersComplete] (http.js:121:23) at Socket.socketOnData [as ondata] (http.js:1587:20) at TCP.onread (net.js:527:27)
```

The URL has multiple redirects.

Unfortunately, I don't know the exact URL, since we are processing 1000s of URLs at a time, and couldn't narrow it down.

Made the follow change to the redirect.js module for line 104:

``` js
try {
    request.uri = url.parse(redirectTo)
} catch(err) {
    return false;
}
```

Thoughts?

### PR DESCRIPTION
**Pull request by cubesapp**
I am getting the following exception when calling request(URL,function(err,res,body){});

```
URIError: URI malformed at decodeURIComponent (native) at Url.parse (url.js:185:19) at Object.urlParse [as parse] (url.js:101:5) at Redirect.onResponse (/myproject/node_modules/request/lib/redirect.js:104:21) at Request.onRequestResponse (/myproject/node_modules/request/request.js:944:22) at ClientRequest.emit (events.js:95:17) at HTTPParser.parserOnIncomingClient [as onIncoming] (http.js:1692:21) at HTTPParser.parserOnHeadersComplete [as onHeadersComplete] (http.js:121:23) at Socket.socketOnData [as ondata] (http.js:1587:20) at TCP.onread (net.js:527:27)
```

The URL has multiple redirects.

Unfortunately, I don't know the exact URL, since we are processing 1000s of URLs at a time, and couldn't narrow it down.

Made the follow change to the redirect.js module for line 104:

``` js
try {
    request.uri = url.parse(redirectTo)
} catch(err) {
    return false;
}
```

Fixes #1864

### CODE REVIEW

**Comment by simov** [comment]
@cubesapp Travis CI is failing, see the logs.

**Comment by rifflock** [comment]
@cubesapp I opened a pull request to your fork to correct the travis issues.

File: lib/redirect.js [hunk 1]
```diff
@@ -101,7 +101,12 @@ Redirect.prototype.onResponse = function (response) {
   }
 
   var uriPrev = request.uri
-  request.uri = url.parse(redirectTo)
+  
+  try{
+	  request.uri = url.parse(redirectTo)
+  } catch(e) {
+	  return false;
```
Comment (FredKSchott) [inline lib/redirect.js]: If it's an invalid URL, it's an invalid URL. This should be an error that gets propagated to the callback via `request.emit('error', ...)` similar to https://github.com/request/request/pull/1873/files#diff-ecf293ae0a8a5d19f25e88fc9e60363aR94.

**Comment by FredKSchott** [comment]
Closing as abandoned (no activity in past year).

Please reopen a new PR based off of master with passing tests if you are still interested in getting this merged and I'll be happy to take a look.

**Pull request closed without merging**